sympy/functions/special/tests/test_error_functions.py

```diff
@@ -338,6 +338,7 @@ def test_ei():
     assert Ei(x).series(x) == EulerGamma + log(x) + x + x**2/4 + \
         x**3/18 + x**4/96 + x**5/600 + O(x**6)
 
+    assert Ei(cos(2)) == -I*pi + Ei(-exp_polar(I*pi)*cos(2))
```

Comment:
Is this valid? `Ei(cos(2))` is real and `unpolarify(-exp_polar(I*pi)*cos(2)) == cos(2)`.

Reply:
It's numerically valid: `N(Ei(cos(2)))` is `-0.676064740123644 + 0.e-20*I` [as expected](https://www.wolframalpha.com/input/?i=ExpIntegralEi(cos(2))).

Reply:
Is there any other way to write this so it doesn't have that explicit `I*pi`? Can't we just return an unchanged `Ei(cos(2))`? I'm not seeing the added value of the `exp_polar` factor...but I don't know a lot about these things, so please forgive any obvious ignorance.

Reply:
We could just get rid of the entire "if not z.is_polar and z.is_negative:" branch that does the recursion, lines 1060-1063. Then Ei(cos(2)) will just evaluate to Ei(cos(2)). This breaks some tests.

Reply:
I'm pretty sure that `Ei(polar_lift(z)) - pi*I` is wrong. `Ei` is a multivalued function, branched at the origin, but it is real-valued on the negative real axis. (It behaves essentially as `log(-z)`.) Hence `Ei(polar_lift(z))` is the same as `Ei(z)`, and there is no need to use polar numbers. (I suspect that the author of these lines was thinking of something like `log(-z) = log(z)  - I*pi` that applies to negative real `z`, but a similar relation is not valid for `Ei`.)

Anyway, I also think that those line should be omitted.